pkg/pulumiyaml/sort.go

```diff
@@ -166,11 +167,18 @@ func topologicallySortedResources(t *ast.TemplateDecl, externalConfig []configNo
 
 		e, ok := intermediates[name.Value]
 		if !ok {
-			diags.Extend(ast.ExprError(name, fmt.Sprintf("resource %q not found", name.Value), ""))
-			return false
+			if isPotentialConfig(name.Value) {
```

Comment:
We will always need to generate the diagnostic. The difference is we will inject a dummy node instead of stopping the algorithm. The caller of `topologicallySortedResources` can then choose to ignore diags (which we will do for eject and get plugins, but nothing else).